pkg/render/goldmane/component.go

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goldmane
+
+import (
+	"fmt"
+
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/securitycontext"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+// The names of the components related to the Guardian related rendered objects.
+const (
+	GoldmaneName               = "goldmane"
+	GoldmaneNamespace          = common.CalicoNamespace
+	GoldmaneServiceAccountName = GoldmaneName
+	GoldmaneDeploymentName     = GoldmaneName
+	GoldmaneRoleName           = GoldmaneName
+	GoldmaneServicePort        = 7443
+	GoldmaneContainerName      = "goldmane"
+
+	GoldmaneKeyPairSecret = "goldmane-key-pair"
+	GoldmaneServiceName   = "goldmane"
+
+	GuardianTargetPort     = 8080
```

Comment:
Do we not want to import these from the Guardian render package?

Reply:
Oh I guess we could do that... I started with goldmane and whisker and if I was to import the values from each other it would be circular dependency. Forgot guardian is in the render package.